src/community/label_propagation.jl

```diff
@@ -0,0 +1,99 @@
+"""
+Community detection using the label propagation algorithm (see [Raghavan et al.](http://arxiv.org/abs/0709.2938)).
+`g`: imput Graph
+`maxiter`: optional, maximum iterations
+return : array containing vertex assignments, the range of the output as 1:countdistinct(membership)
+"""
+function label_propagation(g::SimpleGraph; maxiter=1000)
+    n = nv(g)
+    membership = collect(1:n)
+    active_nodes = Set(vertices(g))
+    nc = NeighComm(collect(1:n), fill(-1,n), 1)
+    i = 0
+    while !isempty(active_nodes) && i < maxiter
+    	i += 1
+        random_order = shuffle(collect(active_nodes))
+        for u in random_order
+          old_comm = membership[u]
+          membership[u] = vote!(g, membership, nc, u)
+            if old_comm != membership[u]
+                for v in out_neighbors(g, u)
+                    push!(active_nodes, v)
+                end
+            else
+                delete!(active_nodes, u)
+            end
+        end
+    end
+    renumber_labels!(membership)
+    membership
+end
+
+"""Fast shuffle Array `a` in UnitRange `r` inplace."""
+function range_shuffle!(r::UnitRange, a::AbstractVector)
+	for i=length(r):-1:2
+		j = rand(1:i)
+		ii = i + r.start - 1
+		jj = j + r.start - 1
+		a[ii],a[jj] = a[jj],a[ii]
+	end
+end
```

Comment:
Is this an implementation of `Base.randperm`?
The Base implementation looks a little more complicated.
https://github.com/JuliaLang/julia/blob/a92c7ff518408182d2bcdf3cbd0dbe3f2c1c1f40/base/random.jl#L1350

I assume that extra bit with the mask is there for correctness.